lib/java/ydoc-polyfill/src/test/java/org/enso/ydoc/api/CallbacksTest.java

```diff
@@ -0,0 +1,107 @@
+package org.enso.ydoc.api;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import org.enso.ydoc.polyfill.ExecutorSetup;
+import org.enso.ydoc.polyfill.web.WebEnvironment;
+import org.graalvm.polyglot.Context;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CallbacksTest extends ExecutorSetup {
+
+  private Context context;
+
+  public CallbacksTest() {}
+
+  public final class TestCallbacks implements MessageCallbacks {
+
+    private Consumer<YjsChannel> handler;
+
+    public TestCallbacks(Consumer<YjsChannel> handler) {
+      this.handler = handler;
+    }
+
+    @Override
+    public void onConnect(YjsChannel channel) {
+      this.handler.accept(channel);
+    }
+  }
+
+  @Before
+  public void setup() throws Exception {
+    super.setup();
+
+    var hostAccess =
+        WebEnvironment.defaultHostAccess
+            // allowImplementations is required to call methods on JS objects from Java,
+            // i.e. to call `YjsChannel::send` in the `TestCallbacks::onConnect` method
+            .allowImplementations(YjsChannel.class)
+            // public access is required to recognize Java lambdas passed to
+            // `YjsChannel::subscribe` method as JS functions.
+            .allowPublicAccess(true)
+            .allowAccess(TestCallbacks.class.getDeclaredMethod("onConnect", YjsChannel.class))
```

Comment:
```suggestion
            // no need for TestCallbacks registration
```
rather use `@HostAccess.Explicit`